### ISSUE TITLE
Documentation for each function and MD documentation

### ISSUE THREAD

**Issue by PEMapModder** [LegendOfMCPE/PocketFactions#10]
Write documentation in the following syntax:

``` php
/**
 * Method details and elaboration in HTML
 * @param ParamType $paramName elaboration
 * @return ReturnType elaboration
*/
```

Do it for every API function (that is supposed that other plugins can use).

**Comment by PEMapModder** [comment]
Assign yourself as "Assignee" if you want to do this.

**Comment by PEMapModder** [comment]
@ijoshuahd is assigned to #10.

**Comment by iJoshuaHD** [comment]
@PEMapModder sorry but i dont know

/**
- Method details and elaboration in HTML
- @param ParamType $paramName elaboration
- @return ReturnType elaboration
  */

stands for? or it just explains how this function works?

i thought i will be working on Documentation xD

**Comment by PEMapModder** [comment]
You can just work on the documentation part :)

**Comment by iJoshuaHD** [comment]
pheww *

**Comment by PEMapModder** [comment]
Lol there is really someone called @return. Must get a lot of notifications ;)
@Param too

**Comment by PEMapModder** [comment]
It explains what the function is for. Only API functions need this, though, since internal functions don't need to get documented.

**Comment by iJoshuaHD** [comment]
yea i dont understand how the documentation thingy works xD

**Comment by PEMapModder** [comment]
Just do the owner guide and player guide then.

Documentation comment is a format. Basically it just explains what the function is for.

**Comment by iJoshuaHD** [comment]
will do :)

**Comment by alejzeis** [comment]
Dis is called java doc.

**Comment by PEMapModder** [comment]
PHP doc it is called in PHPStorm.

**Comment by alejzeis** [comment]
It's basically a copy of java doc...

On Tuesday, July 22, 2014, PEMapModder notifications@github.com wrote:

> PHP doc it is called in PHPStorm.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/LegendOfMCPE/PocketFactions/issues/10#issuecomment-49822986
> .

##

**Comment by PEMapModder** [comment]
I don't think the `|` thing is copied.

**Comment by alejzeis** [comment]
Where's that?

On Tuesday, July 22, 2014, PEMapModder notifications@github.com wrote:

> I don't think the | thing is copied.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/LegendOfMCPE/PocketFactions/issues/10#issuecomment-49828260
> .

##

**Comment by PEMapModder** [comment]
@jython234 

``` php
/**
 * @param string|Player $player
 * @return string
 */
public function getPlayerFile($player){
    if($player instanceof Player) $player = $player->getName();
    return $this->getDataFolder().strtolower($player).".yml";
}
```

**Comment by swagking47** [comment]
@PEMapModder PEMapModder does every player have his own file ?

**Comment by alejzeis** [comment]
I guess your right, but still those tags and the doc format look like java
doc to me...

On Tuesday, July 22, 2014, swagking47 notifications@github.com wrote:

> PEMapModder does every player have his own file ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/LegendOfMCPE/PocketFactions/issues/10#issuecomment-49828693
> .

##

**Comment by PEMapModder** [comment]
@swagking47 just an example.

**Comment by caingaroo** [comment]
Thee plugin has a problem in the Faction ranks
[Server] 19:33:13 [INFO] Loading DevTools v1.10.0
[Server] 19:33:18 [INFO] Enabling DevTools v1.10.0
[Server] 19:33:18 [INFO] Loading source plugin PocketFactions v2.0.0
[Server] 19:33:18 [INFO] [DevTools] Registered folder plugin loader
[Server] 19:33:18 [INFO] Preparing level "Arena"
[Server] 19:33:18 [INFO] Enabling PocketFactions v2.0.0
[Server] 19:33:18 [NOTICE] ArrayOutOfBoundsException: "Undefined index: permissions" (E_NOTICE) in "/PocketFactions-new/src/pocketfactions/faction/FactionRank" at line 32
[Server] 19:33:18 [INFO] Disabling PocketFactions v2.0.0
[Server] 19:33:19 [INFO] Starting GS4 status listener
[Server] 19:33:19 [INFO] Setting query port to 19132
[Server] 19:33:19 [INFO] Query running on 0.0.0.0:19132
[Server] 19:33:19 [INFO] Default game type: SURVIVAL
[Server] 19:33:19 [INFO] Done (13.005s)! For help, type "help" or "?"

**Comment by PEMapModder** [comment]
@caingaroo plugin is not finished yet.

**Comment by caingaroo** [comment]
@PEMapModders why didnt you labble it on the plugin , or did i miss it somehoe 😯😯

**Comment by PEMapModder** [comment]
If the plugin is completed, we would make a release.
Where do you think we would label it?

**Comment by caingaroo** [comment]
Dont know, but could you notify  me win the plugin is finished

**Comment by alejzeis** [comment]
Ehh doxygen.

**Comment by PEMapModder** [comment]
We will create a release when a beta is ready, and submit on forums when initial release is ready.

**Comment by jython234** [comment]
@caingaroo compile it yourself :).

On Sunday, February 22, 2015, caingaroo <notifications@github.com> wrote:

> Dont know, but could you notify me win the plugin is finished
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LegendOfMCPE/PocketFactions/issues/10#issuecomment-75447872>
> .
>


--

**Comment by PEMapModder** [comment]
@jython234 you don't need to compile it. It can load source plugin.

**Comment by astersx** [comment]
will it be compatible with customRanks plugin?

**Comment by PEMapModder** [comment]
@astersx how are they related?

**Comment by astersx** [comment]
Hmmm,  customRanks hide ProFactions name,  So [faction]name,  with custom ranks,  [member]name

**Comment by iksaku** [comment]
They are not related at all..

**Comment by astersx** [comment]
Okok!!!  Sooo,  Profaction work?

**Comment by astersx** [comment]
sorry pocketfactions

**Comment by PEMapModder** [comment]
@astersx see #1

**Comment by PEMapModder** [comment]
@astersx 
1. It is supposed to support as many economy plugins as possible, although @onebone's economys and @LegendOfMCPE's xEcon (W.I.P.) will have priority.
2. If CustomRanks overrides other plugins' tags, it has a problem. Don't use it.
3. Why do you even use Economys Land with Factions? That's unreasonable.
4. DIsable commands? Just revoke the permissions with a permissions plugin.
5. If you need to avoid making multiple accounts, maybe you should use an auth plugin that checks that, e.g. ServerAuth by EvolSoft.

**Comment by astersx** [comment]
You right :+1: ,  + if i install another auth plugin,  all accounts will be lost T.T

**Comment by astersx** [comment]
And it works?

**Comment by PEMapModder** [comment]
@astersx requested @evolsoft to support that?

**Comment by astersx** [comment]
To support what?

**Comment by PEMapModder** [comment]
Transferring data.

**Comment by astersx** [comment]
I dont know,  i didnt check another auth plugins

**Comment by astersx** [comment]
Every plugin devs have theyr own codes,  so im sure i cant transfer data!!!!

**Comment by PEMapModder** [comment]
Ref: EvolSoft/ServerAuth#44

**Comment by astersx** [comment]
I see,  so i can manually, :|,  i can zip all and unpack on new plugin,  or rename simpleauth to serverauth and serverauth official to test,  then copy files from test in server auth,  so i dont have to move 10.000 files. But is a lot of headache!!!

anyway u didnt reply,  pocket faction work? i can use?

**Comment by PEMapModder** [comment]
@astersx nope, they use different hash algos, so you can't do it without having an update on ServerAuth.

**Comment by astersx** [comment]
I see,  and i use .yml as database so its not possible,  since serverauth love mysql

**Comment by PEMapModder** [comment]
What is this? https://github.com/EvolSoft/ServerAuth/blob/master/ServerAuth/resources/config.yml#L30

Also, please don't use the issue tracker as a chatroom. Get on our IRC channel or Gitter chatroom if you need further support.
